source/includes/indexes/indexes.c

```diff
@@ -149,26 +149,78 @@ main (void)
         // end-create-search-index
     }
     {
-        // start-create-search-indexes
-        bson_t cmd;
-        bson_error_t error;
-        char *cmd_str = bson_strdup_printf (
-            BSON_STR ({
-                "createSearchIndexes" : "%s",
-                "indexes" : [ {"definition" : {"mappings" : {"dynamic" : false}}, "name" : "<first index name>"},
-                              {"definition" : {"mappings" : {"dynamic" : false}}, "name" : "<second index name>"} ]
-            }),
-            "<collection name>");
-        bson_init_from_json (&cmd, cmd_str, -1, &error);
-        bson_free (cmd_str);
+      // start-create-vector-search-index
+      bson_t cmd;
+      bson_error_t error;
+      char * cmd_str = bson_strdup_printf(
+        BSON_STR({
+          "createSearchIndexes": "%s",
+          "indexes": [{
+            "name": "<index name>",
+            "type": "vectorSearch",
+            "definition": {
+              "fields": [{
+                "type": "vector",
+                "path": "plot_embedding",
+                "numDimensions": 1536,
+                "similarity": "euclidean"
+              }]
+            }
+          }]
+        }),
+        "<collection name>"
+      );
+      bson_init_from_json( & cmd, cmd_str, -1, & error);
```

Comment:
Minor: The spacing around the unary ampersand `&` is unusual.
```suggestion
      bson_init_from_json(&cmd, cmd_str, -1, &error);
```